ledger/participant-integration-api/src/main/scala/platform/apiserver/meteringreport/HmacSha256.scala

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.platform.apiserver.meteringreport
+
+import spray.json.DefaultJsonProtocol._
+import spray.json.{JsValue, JsonFormat, RootJsonFormat, deserializationError}
+
+import java.util.Base64
+import javax.crypto.spec.SecretKeySpec
+import javax.crypto.{KeyGenerator, Mac}
+import scala.util.Try
+
+object HmacSha256 {
+
+  def toBase64(bytes: Array[Byte]): String = Base64.getUrlEncoder.encodeToString(bytes)
+  def fromBase64(base64: String): Either[Throwable, Array[Byte]] = Try(
+    Base64.getUrlDecoder.decode(base64)
+  ).toEither
+
+  // The key used for both mac and key generation as defined in
+  // https://docs.oracle.com/javase/9/docs/specs/security/standard-names.html
+  val algorithm = "HmacSHA256"
+
+  def compute(key: Key, message: Array[Byte]): Either[Throwable, Array[Byte]] = {
+    Try {
+      val mac = Mac.getInstance(algorithm)
```

Comment:
Not sure it is applicable here, but creating an instance for each compute might be very expensive, but reusing an instance in a highly concurrent environment might be slow because of the problem in jdk. You could consider using com.daml.crypto.MacPrototype instead.

Reply:
Yes, I forgot about that work.  Nice idea, switched.